language-service/src/services/yamlCompletion.ts

```diff
@@ -85,8 +87,17 @@ export class YAMLCompletion {
 			nodeEndPos.character += 1;
 			overwriteRange = Range.create(nodeStartPos, nodeEndPos);
 		}else if (node && (node.type === 'string' || node.type === 'number' || node.type === 'boolean')) {
-			//console.log('type = string | nuber | boolean');
-			overwriteRange = Range.create(document.positionAt(node.start), document.positionAt(node.end));
+            //console.log('type = string | nuber | boolean');
+            const startPosition = document.positionAt(node.start);
+            let endPosition = document.positionAt(node.end);
+
+            // when start and end positions are not on the same line and node is a temporary holder then
+            // we must have misplaced the end of the range one line below the start of the range
+            // because of the mismatch in line length between the temporary currentDoc with holder and actual document.
+            if (startPosition.line + 1 === endPosition.line && endPosition.character === 0 && node.location === nodeHolder) {
```

Comment:
```suggestion
startPostiion.line < endPosition.line
```

Reply:
No, I want to be as specific as possible here. The end should be exactly on the next line to the case to be detected